utils/utils.go

```diff
@@ -352,3 +352,28 @@ func ReadFile(filePath string) ([]byte, error) {
 	}
 	return bytes, nil
 }
+
+// IsYAMLFile returns true if the given file path is a valid YAML file.
+func IsYAMLFile(filePath string) bool {
+	fileInfo, err := os.Stat(filepath.Clean(filePath))
+	if err != nil {
+		return false
+	}
+	isDir := fileInfo.IsDir()
+	hasYAMLExtension := strings.HasSuffix(filePath, ".yaml") || strings.HasSuffix(filePath, ".yml")
+	return !isDir && hasYAMLExtension
+}
+
+// FindFileInDir finds and returns the path of the given file name in the given directory.
+func FindFileInDir(dirPath, fileName string) (string, error) {
+	files, err := os.ReadDir(filepath.Clean(dirPath))
+	if err != nil {
+		return "", fmt.Errorf("error reading directory %q : %w", dirPath, err)
+	}
+	for _, f := range files {
+		if !f.IsDir() && f.Name() == fileName {
+			return filepath.Join(dirPath, fileName), nil
+		}
+	}
```

Comment:
why list files and iterate through them? 
Join the dirPath and defaultRunTemplateFileName and see if such a file exists?

Reply:
aah..right.